housekeeping/makeBuild.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -xe
+
+BUILD_DIR="${BUILD_DIR:?BUILD_DIR variable is not defined}"
+BUILD_TARGET="${BUILD_TARGET:-test}"
+BUILD_TREADS="${BUILD_TREADS:-4}"
```

Comment:
We can calculate the default value automatically
```
make -j$((`nproc`+1))
```

Reply:
did not work for mac, same as cache we will do it next CI related pr.